react-components/src/components/Architecture/DomainObjectPanel.tsx

```diff
@@ -37,16 +37,16 @@ export const DomainObjectPanel = (): ReactElement => {
   useEffect(() => {
     DomainObjectPanelUpdater.setDomainObjectDelegate(setCurrentDomainObjectInfo);
 
-    if (commands === undefined || info === undefined) {
-      return;
-    }
-
     // Set in the get string on the copy command if any
+  }, [setCurrentDomainObjectInfo, commands]);
+
+  // Fore the getString to be updated
+  if (commands !== undefined && info !== undefined) {
     for (const command of commands) {
       if (command instanceof CopyToClipboardCommand)
         command.getString = () => toString(info, t, unitSystem);
     }
-  }, [setCurrentDomainObjectInfo, commands]);
+  }
 
```

Comment:
What is this change for? I think the for-loop should still be inside the useEffect, as it seems like something that's unnecessary to do on every render. But Perhaps `unitSystem` should be in the dependency array?

Reply:
The promblem vas that the getString wasn't apply and the button disabled. I dont know why, but when I changed the code, it worked.